### ISSUE TITLE
Triton Crashes with onnxruntime error

### ISSUE THREAD

**Issue by sourabh-burnwal** [triton-inference-server/server#4394]
I have an ensemble of 10 models in my model repository. The models are a mixture of python, torchscript, dali, and onnx backends. I am using triton in RTX 3050, with 8gb VRAM, 16gb of RAM, and i7 11th gen processor. When the server is up and I try to get the inferences using a client script, the server crashes on the first hit. This has happened in many other scenarios as well. The conclusion I came to is, there is a limit of number of models we can load and use for a particular hardware specifications. But I need more clarity on why the server crashes and if there is any solution or workaround for this. 

Sharing the error:
`
2022-05-17 20:23:16.005702183 [E:onnxruntime:, sequential_executor.cc:339 Execute] Non-zero status code returned while running Conv node. Name:'Conv_16' Status Message: /workspace/onnxruntime/onnxruntime/core/framework/bfc_arena.cc:330 void* onnxruntime::BFCArena::AllocateRawInternal(size_t, bool) Failed to allocate memory for requested buffer of size 1115955200
`

So, basically two questions:
1. Why it happens and how to prevent it?
2. How to decide the hardware specs based on the number and kind of models one has?
Thanks!!

**Comment by krishung5** [comment]
Hi @sourabh-burnwal, regarding the error message, I think you need to adjust the size of the shared memory using `--shm-size` flag to make sure that it is large enough for your use case. For your second question, I think you can try to monitor system usage when running Triton with those models to get a better sense of configuring hardware specs. Using [Metrics](https://github.com/triton-inference-server/server/blob/main/docs/metrics.md) might be helpful as well for getting GPU statistics.

**Comment by GuanLuo** [comment]
Since you are running an ensemble, can you try to break it down and manually run the models one by one? Based on the error message, 1 GB of memory is being requested and it seems to be large, I wonder if the model is being executed with invalid input.

And I also wonder if the server was terminated by the OS due to OOM?

**Comment by Tabrizian** [comment]
ORT backend recently added a setting to give more control over the memory pool. The setting name is `memory.enable_memory_arena_shrinkage` which might be helpfu. Also, the pytorch backend added ENABLE_CACHE_CLEANING option which should also help with the memory usage.

Closing due to in-activity.
